app/services/csv/lettings_log_csv_service.rb

```diff
@@ -177,12 +177,20 @@ def value(attribute, log)
           value
         when "labels"
           answer_label = get_label(value, attribute, log)
-          answer_label || label_if_boolean_value(value) || value
+          answer_label || label_if_boolean_value(value) || (YES_OR_BLANK_ATTRIBUTES.include?(attribute) && value != 1 ? nil : value)
         end
       end
     end
 
     def get_label(value, attribute, log)
+      return LETTYPE_LABELS[value] if attribute == "lettype"
+      return IRPRODUCT_LABELS[value] if attribute == "irproduct"
+      return LAR_LABELS[value] if attribute == "lar"
+      return NEWPROP_LABELS[value] if attribute == "newprop"
+      return INCREF_LABELS[value] if attribute == "incref"
+      return conventional_yes_no_label(value) if CONVENTIONAL_YES_NO_ATTRIBUTES.include?(attribute)
+      return "Yes" if YES_OR_BLANK_ATTRIBUTES.include?(attribute) && value == 1
+
```

Comment:
maybe a switch statement, or nested labels like `LABELS = {lettype: LETTYPE_LABELS, irproduct: IRPRODUCT_LABELS ... }` and `return LABELS[attribute][value] if LABELS[attribute].present?` or something like that. But only if it reads easier, otherwise this is fine

Reply:
yaeh I like the nested labels, have done, good idea!